src/subcommand/torrent/announce.rs

```diff
@@ -0,0 +1,135 @@
+use crate::common::*;
+
+const URI_HELP: &str = "Announce an infohash and list the response (a compact peer list).";
+
+const INPUT_HELP: &str =
+  "Generate a compact peer list from a metainfo at `INPUT`. If `INPUT` is `-`, read \
+                          metainfo from standard input.";
+
+const INPUT_FLAG: &str = "input-flag";
+
+const INPUT_POSITIONAL: &str = "<INPUT>";
+
+#[derive(StructOpt)]
+#[structopt(
+  help_message(consts::HELP_MESSAGE),
+  version_message(consts::VERSION_MESSAGE),
+  about(URI_HELP)
+)]
+pub(crate) struct Announce {
+  #[structopt(
```

Comment:
Thanks for replicating this input flag/positional stuff. It's annoying, but I think it makes the CLI nicer.

Reply:
There's a simple solution that's implemented by at least one widely used tool: no positional arguments---everything gets a flag.

Reply:
That's what I did initially, but users complained that it was too verbose.